stdlib/public/Distributed/DistributedActor.swift

```diff
@@ -275,6 +303,28 @@ public protocol DistributedActor: AnyActor, Identifiable, Hashable
 
 }
 
+#if compiler(>=5.9.0) // since this compiler version, we synthesize the `localUnownedExecutor` instead.
+@available(SwiftStdlib 5.7, *)
+extension DistributedActor {
+  @available(SwiftStdlib 5.7, *)
+  public nonisolated var unownedExecutor: UnownedSerialExecutor {
+    if #available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *) {
+    // In these versions, we moved to asking developers to implement
+    // `localUnownedExecutor` and as such delegate to it. The `??` fallback
+    // is likely to never be triggered, but technically could be if someone queried
+    // the `unownedExecutor` property directly on a default distributed remote actor reference,
+    // in previous language versions, so we keep these semantics for compatibility.
+    return self.localUnownedExecutor ??
+        UnownedSerialExecutor(Builtin.buildDefaultActorExecutorRef(self))
+    } else {
+      // pessimistic reimplementation of the semantics the synthesized body would have had.
+      // semantically, this is the only implementation possible on older Swift's here.
+      return UnownedSerialExecutor(Builtin.buildDefaultActorExecutorRef(self))
+    }
+  }
+}
+#endif // swift 5.9
```

Comment:
Is this dance correct?

Basically, previously we never had this property `unownedExcutor` _at all_ on distributed actors, but we'd just treat them all as default actors (in a check in `LowerHopToActor`).

Now in 5.9 we gain this method and synthesize the `localUnownedExecutor -> SerialUnownedExecutor?` however in order to not have to rewrite all lowering and rest of actor runtime which I'm sure we'd miss something someday, we added the `unownedExecutor` -- so the runtime can call into that. Practically speaking, code isolated on a remote instance does not exist (!), so this would never be triggered in such situation.

Questions:

- Should we **NOT** add this method and instead rewrite `LowerHopToActor` to be aware of the optional `localUnownedExecutor` and if it returned `nil` just crash the process?
- If we SHOULD keep this method, is this the right way to implement a shim?

Reply:
Turns out removing this "shim" is very painful, because I cannot switch over the Optional easily in `LowerHopToActor` since we can't even introduce basic blocks there to cause a crash if it was `nil` etc.

Perhaps the right solution here is to special handle the distributed NON-default actor by having a path in `LowerHopToActor` which calls an `unwrapUnownedExecutor` on the distributed actor and we make it NOT be a customization point 🤔 

Open to ideas @jckarter @kavon @rjmccall @DougGregor